types/fbt/index.d.ts

```diff
@@ -1,171 +1,285 @@
+     */
+    capitalize?: boolean;
+}
 
-    // https://github.com/facebook/fbt/blob/e9c591f451dbfc91852e316869ae39ad41848c55/runtime/nonfb/IntlVariations.js#L9-L21
-    interface IntlVariations extends IntlVariationsGender {
-        BITMASK_NUMBER: 28;
-        BITMASK_GENDER: 3;
-        NUMBER_ZERO: 16;
-        NUMBER_ONE: 4;
-        NUMBER_TWO: 8;
-        NUMBER_FEW: 20;
-        NUMBER_MANY: 12;
-        NUMBER_OTHER: 24;
-    }
+export interface PluralOptions {
+    /**
+     * Represents the plural form of the string in English. Default is `{singular} + 's'`
+     */
+    many?: string;
+    /**
+     * `"yes"|"no"|"ifMany"`: Whether to show the `{number}` in the string.
+     *
+     * > Note that the singular phrase never has a token, but inlines to `1`.
+     * ? This is to account for languages like Hebrew for which showing the actual number isn't appropriate
+     *
+     *  - "no": (DEFAULT) Don't show the count
+     *  - "ifMany": Show the count only in plural case
+     *  - "yes": Show the count in all cases
+     */
+    showCount?: 'yes' | 'no' | 'ifMany';
+    /**
+     * Name of the token where count shows up. (Default: `"number"`)
+     */
+    name?: string;
+    /**
+     * For overriding the displayed number
+     */
+    value?: unknown;
+}
 
-    // https://github.com/facebook/fbt/blob/c2d363a40b622d5aaf80ff1d249b38604fd869f6/transform/babel-plugin-fbt/FbtConstants.js#L22-L27
-    type PronounType = 'object' | 'possessive' | 'reflexive' | 'subject';
+export interface FbtOptions {
+    /**
+     * Text author
+     */
+    author?: string;
+    /**
+     *  Use a "common" string repository
+     *  To use the strings at runtime, there is the fbt.c(...) function call or the <fbt common=true>...</fbt> JSX API.
+     *  NOTE: The transform will throw if it encounters a common string not in the map provided.
+     *  See: https://facebook.github.io/fbt/docs/common#runtime-api
+     */
+    common?: boolean;
+    /**
+     *  Informs [collection](https://facebook.github.io/fbt/docs/collection/) to skip this string (useful for tests/mocks)
+     */
+    doNotExtract?: boolean;
+    /**
+     *  (Default: false) FBT normally consolidates whitespace down to one space (' '). Turn this off by setting this to true
+     */
+    preserveWhitespace?: boolean;
+    /**
+     * Project to which the text belongs
+     */
+    project?: string;
+    /**
+     * IntlVariations.GENDER_*: Pass an [implicit subject](https://facebook.github.io/fbt/docs/implicit_params/) gender to a partially formed text
+     */
+    subject?: IntlVariationsGender;
+}
 
-    type IntlVariationsGenderValues = $Values<IntlVariationsGender>;
+export interface Output {
+    fbt: string[];
+    params: string;
+}
 
-    interface IntlViewerContext {
-        GENDER: IntlVariationsGenderValues;
-        locale: string;
-    }
+export interface Fbt {
+    (text: string, description: string, options?: FbtOptions): Output['fbt'];
```

Comment:
@retyui, you've specified the output here as `string[]`, but it should be `string`, right? CC @yrichard.

Reply:
Ah, sorry, I can't easily check right now because I'm traveling, but also, I dropped my involvement with FBT, due to the lack of support from Facebook when in conjunction with Typescript... Sorry!